internal/provider/clickhouse_data_source.go

```diff
@@ -223,7 +223,10 @@ func (d *ClickhouseDataSource) Read(ctx context.Context, req datasource.ReadRequ
 		}
 	}
 
-	response, err := d.svc.Get(ctx, &clickhouse.GetClusterRequest{ClusterId: data.Id.ValueString()})
+	response, err := d.svc.Get(ctx, &clickhouse.GetClusterRequest{
+		ClusterId: data.Id.ValueString(),
+		Sensitive: true,
```

Comment:
Without this field API wouldn't return password, just empty string. It's safe to call it every time bevause if user have no permissions to access secrets it would be treatet like sensitive=false